wgpu-hal/src/gles/device.rs

```diff
@@ -237,12 +239,31 @@ impl super::Device {
             binding_array: BoundsCheckPolicy::Unchecked,
         };
 
+        let update_naga_options = primitive_topology == Some(wgt::PrimitiveTopology::PointList)
+            && naga_stage == naga::ShaderStage::Vertex;
+        // Update naga options if mesh consist of point list and it is vertex shader
+        let naga_options = if update_naga_options {
+            let mut wrt_flags = context.layout.naga_options.writer_flags;
+            wrt_flags.set(glsl::WriterFlags::FORCE_POINT_SIZE, true);
+            glsl::Options {
+                version: context.layout.naga_options.version,
+                writer_flags: wrt_flags,
+                binding_map: context.layout.naga_options.binding_map.clone(),
+                zero_initialize_workgroup_memory: context
+                    .layout
+                    .naga_options
+                    .zero_initialize_workgroup_memory,
+            }
+        } else {
+            context.layout.naga_options.clone()
+        };
```

Comment:
Looking at the Vulkan backend,

https://github.com/gfx-rs/wgpu/blob/48d8666de002c7bfddbe85a2596f7748b4cc18b4/wgpu-hal/src/vulkan/adapter.rs#L1256-L1262

I'm not sure what the best solution is but cloning the options here doesn't sound too good to me.

Are there any issues if we always set `FORCE_POINT_SIZE` to true?

Reply:
@teoxoy I don't think I understand the domain well enough to comment.

@cwfitzgerald what do you think?

Reply:
I could rewrite to something like this (extracting to reusable method):
```rust
let update_naga_options = primitive_topology == Some(wgt::PrimitiveTopology::PointList)
    && naga_stage == naga::ShaderStage::Vertex;
// Update naga options if mesh consist of point list and it is vertex shader
let reflection_info = if update_naga_options {
    let mut wrt_flags = context.layout.naga_options.writer_flags;
    wrt_flags.set(glsl::WriterFlags::FORCE_POINT_SIZE, true);
    let options = glsl::Options {
        version: context.layout.naga_options.version,
        writer_flags: wrt_flags,
        binding_map: context.layout.naga_options.binding_map.clone(),
        zero_initialize_workgroup_memory: context
            .layout
            .naga_options
            .zero_initialize_workgroup_memory,
    };
    let mut writer = glsl::Writer::new(
        &mut output,
        &shader.module,
        &shader.info,
        &options,
        &pipeline_options,
        policies,
    )
    .map_err(|e| {
        let msg = format!("{e}");
        crate::PipelineError::Linkage(map_naga_stage(naga_stage), msg)
    })?;
    writer.write().map_err(|e| {
        let msg = format!("{e}");
        crate::PipelineError::Linkage(map_naga_stage(naga_stage), msg)
    })
} else {
    let mut writer = glsl::Writer::new(
        &mut output,
        &shader.module,
        &shader.info,
        &context.layout.naga_options,
        &pipeline_options,
        policies,
    )
    .map_err(|e| {
        let msg = format!("{e}");
        crate::PipelineError::Linkage(map_naga_stage(naga_stage), msg)
    })?;
    writer.write().map_err(|e| {
        let msg = format!("{e}");
        crate::PipelineError::Linkage(map_naga_stage(naga_stage), msg)
    })
}?;
```

Reply:
@teoxoy both `Options` and `PipelineOptions` are from naga. It requires larger refactoring. 

Originally I implemented the parameter not as part of `WriterFlags`, but `PipelineOptions` (https://github.com/gfx-rs/naga/commit/4df4161d8d00c82910cb9cbdb8599abd1f82af46#diff-cc33bd3c8c18000d3ee7b6f5578e8fa07ffd08d87b0aa3d9525a107937fb3d3fR264). I followed how it was implemented for Metal [metal/device.rs#L71](https://github.com/gfx-rs/wgpu/blob/master/wgpu-hal/src/metal/device.rs#L71) [back/msl/mod.rs#L238 ](https://github.com/gfx-rs/naga/blob/master/src/back/msl/mod.rs#L238)

The same functionality is implemented in two ways.

Reply:
I don't think it's the same functionality, for metal it works like a filter and is needed because you'll get an error otherwise.

For Vulkan, we can always force naga to write it and will be ignored by the driver if it's not a point list primitive.

I think we can have gles behave the same as Vulkan here. Let's see what @cwfitzgerald thinks as well.

Reply:
I think why we need to have `allow_point_size` for metal is due to:

> glslang generates gl_PerVertex struct with gl_CullDistance, gl_ClipDistance and gl_PointSize builtin inside by default

(from https://github.com/gfx-rs/naga/blob/37be4df2b8e7c56b0390445f4457955211040433/src/back/wgsl/writer.rs#L1161-L1169)

And if we always set `FORCE_POINT_SIZE` to true, we'll essentially behave the same as glslang.

Reply:
As per discussion, I'm setting `FORCE_POINT_SIZE` to make it consistent with the Vulkan backend.